sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala

```diff
@@ -349,7 +349,7 @@ private[sql] final case class FieldReference(parts: Seq[String]) extends NamedRe
 
 private[sql] object FieldReference {
   def apply(column: String): NamedReference = {
-    LogicalExpressions.parseReference(column)
+    LogicalExpressions.parseReference("`" + column + "`")
```

Comment:
I think we need to fix the caller side. We shouldn't call `FieldReference.apply(String)` which parses the given string. We should call `FieldReference(Seq(col_name))`.